spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkContentFiles.java

```diff
@@ -161,40 +173,132 @@ private void checkSparkDataFile(Table table) throws IOException {
         "The number of files should match", dataFiles.size(), sparkDataFiles.size());
 
-    Types.StructType dataFileType = DataFile.getType(table.spec().partitionType());
+    Types.StructType dataFileType = DataFile.getType(dataFilesSpec.partitionType());
     StructType sparkDataFileType = sparkDataFiles.get(0).schema();
-    SparkDataFile wrapper = new SparkDataFile(dataFileType, sparkDataFileType);
+    SparkDataFile dataFileWrapper = new SparkDataFile(dataFileType, sparkDataFileType);
 
     for (int i = 0; i < dataFiles.size(); i++) {
-      checkDataFile(dataFiles.get(i), wrapper.wrap(sparkDataFiles.get(i)));
+      checkDataFile(dataFiles.get(i), dataFileWrapper.wrap(sparkDataFiles.get(i)));
+    }
+
+    Dataset<Row> positionDeleteFileDF =
+        spark.read().format("iceberg").load(tableLocation + "#delete_files").where("content = 1");
+    List<Row> sparkPositionDeleteFiles = shuffleColumns(positionDeleteFileDF).collectAsList();
+    Assert.assertEquals(
+        "The number of files should match",
+        positionDeleteFiles.size(),
+        sparkPositionDeleteFiles.size());
+
+    Types.StructType positionDeleteFileType = DataFile.getType(dataFilesSpec.partitionType());
+    StructType sparkPositionDeleteFileType = sparkPositionDeleteFiles.get(0).schema();
+    SparkDeleteFile positionDeleteFileWrapper =
+        new SparkDeleteFile(positionDeleteFileType, sparkPositionDeleteFileType);
+
+    for (int i = 0; i < positionDeleteFiles.size(); i++) {
+      checkDeleteFile(
+          positionDeleteFiles.get(i),
+          positionDeleteFileWrapper.wrap(sparkPositionDeleteFiles.get(i)));
+    }
+
+    Dataset<Row> equalityDeleteFileDF =
+        spark.read().format("iceberg").load(tableLocation + "#delete_files").where("content = 2");
+    List<Row> sparkEqualityDeleteFiles = shuffleColumns(equalityDeleteFileDF).collectAsList();
+    Assert.assertEquals(
+        "The number of files should match",
+        equalityDeleteFiles.size(),
+        sparkEqualityDeleteFiles.size());
+
+    Types.StructType equalityDeleteFileType = DataFile.getType(table.spec().partitionType());
+    StructType sparkEqualityDeleteFileType = sparkEqualityDeleteFiles.get(0).schema();
+    SparkDeleteFile equalityDeleteFileWrapper =
+        new SparkDeleteFile(equalityDeleteFileType, sparkEqualityDeleteFileType);
+
+    for (int i = 0; i < equalityDeleteFiles.size(); i++) {
+      checkDeleteFile(
+          equalityDeleteFiles.get(i),
+          equalityDeleteFileWrapper.wrap(sparkEqualityDeleteFiles.get(i)));
     }
   }
 
+  private Dataset<Row> shuffleColumns(Dataset<Row> df) {
+    List<Column> columns =
+        Arrays.stream(df.columns()).map(ColumnName::new).collect(Collectors.toList());
+    Collections.shuffle(columns);
+    return df.select(columns.toArray(new Column[0]));
+  }
+
   private void checkDataFile(DataFile expected, DataFile actual) {
+    Assert.assertEquals("Content must match", expected.content(), actual.content());
```

Comment:
nit: we can have `checkContentFile` and extract common assertions into that

Reply:
Yeah, I should have done that.

Reply:
Fixed.